distributed/dashboard/components/scheduler.py

```diff
@@ -1720,6 +1724,336 @@ def __del__(self):
+
+            return nodes_layout, arrows_layout
+
+    @without_property_validation
+    def update(self):
+
+        if not self.scheduler.task_groups:
+            self.subtitle.text = "Scheduler is empty."
+        else:
+            self.subtitle.text = " "
+
+        if self.nodes_layout.keys() != self.scheduler.task_groups.keys():
+            self.nodes_layout, self.arrows_layout = self.update_layout()
+
+        nodes_data = {
+            "x": [],
+            "y": [],
+            "name": [],
+            "name_short": [],
+            "color": [],
+            "tot_tasks": [],
+            "x_start": [],
+            "x_end": [],
+            "y_start": [],
+            "y_end": [],
+            "x_end_progress": [],
+            "progress": [],
+            "x_label": [],
+            "y_label": [],
+            "mem_alpha": [],
+            "node_line_width": [],
+            "xc_pbar": [],
+            "yc_pbar": [],
+            "comp_tasks": [],
+        }
+
+        arrows_data = {
+            "xs": [],
+            "ys": [],
+            "xe": [],
+            "ye": [],
+        }
+
+        for key, tg in self.scheduler.task_groups.items():
+            x = self.nodes_layout[key]["x"]
+            y = self.nodes_layout[key]["y"]
+
+            comp_tasks = (
+                tg.states["waiting"] + tg.states["no-worker"] + tg.states["processing"]
+            )
+            tot_tasks = sum(tg.states.values())
+
+            # main boxes layout
+            nodes_data["x"].append(x)
+            nodes_data["y"].append(y)
+
+            nodes_data["name"].append(tg.prefix.name)
+            nodes_data["name_short"].append(
+                tg.prefix.name[:10]
+            )  # This needs to be different
+
+            nodes_data["x_label"].append(x - self.width_node / 2 + 0.2)
+            nodes_data["y_label"].append(y + self.height_node / 2 - 0.2)
+
+            nodes_data["color"].append(color_of(tg.prefix.name))
+            nodes_data["tot_tasks"].append(tot_tasks)
+
+            # memory alpha factor by 0.4 if not get's too dark
+            nodes_data["mem_alpha"].append(
+                (tg.states["memory"] / sum(tg.states.values())) * 0.4
+            )
+
+            # main box line width
+            if tg.states["processing"]:
+                nodes_data["node_line_width"].append(5)
+            else:
+                nodes_data["node_line_width"].append(1)
+
+            # progress bar data update
+            Lbar = self.width_node - 0.2
+            nodes_data["x_start"].append(x - self.width_node / 2 + 0.1)
+            nodes_data["x_end"].append(x - self.width_node / 2 + 0.1 + Lbar)
+
+            Hbar = 2
+            nodes_data["y_start"].append(y - self.height_node / 2 + 0.1)
+            nodes_data["y_end"].append(y - self.height_node / 2 - 0.1 + Hbar)
+
+            completed = 1.0 - (comp_tasks / tot_tasks)
+
+            nodes_data["progress"].append(completed * 100)
```

Comment:
We can drop the `1.0 -` part here since `completed` should be the fraction of completed tasks 

```suggestion
            completed = comp_tasks / tot_tasks
```